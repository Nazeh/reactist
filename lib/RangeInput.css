.reactist.range_input .range_btn {
  height: 30px;
  width: 30px;
  background-size: 20px;
  display: inline-block;
  background-position: center;
  background-repeat: no-repeat;
}
.reactist.range_input .range_btn.plus {
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' width='23' height='23' viewBox='0 0 23 23'%3E %3Cdefs%3E %3Cpath id='a' d='M0 0h23v23H0z'/%3E %3C/defs%3E %3Cg fill='none' fill-rule='evenodd'%3E %3Cmask id='b' fill='%23fff'%3E %3Cuse xlink:href='%23a'/%3E %3C/mask%3E %3Cpath fill='%23979797' fill-rule='nonzero' d='M11.5 21a9.5 9.5 0 1 0 0-19 9.5 9.5 0 0 0 0 19zm0 1C5.701 22 1 17.299 1 11.5S5.701 1 11.5 1 22 5.701 22 11.5 17.299 22 11.5 22z' mask='url(%23b)'/%3E %3Crect width='2' height='9' x='10.5' y='7' fill='%23979797' mask='url(%23b)' rx='1' transform='rotate(90 11.5 11.5)'/%3E %3Crect width='2' height='9' x='10.5' y='7' fill='%23979797' mask='url(%23b)' rx='1' transform='rotate(-180 11.5 11.5)'/%3E %3C/g%3E %3C/svg%3E");
}
.reactist.range_input .range_btn.minus {
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' width='23' height='23' viewBox='0 0 23 23'%3E %3Cdefs%3E %3Cpath id='a' d='M0 0h23v23H0z'/%3E %3C/defs%3E %3Cg fill='none' fill-rule='evenodd'%3E %3Cmask id='b' fill='%23fff'%3E %3Cuse xlink:href='%23a'/%3E %3C/mask%3E %3Crect width='2' height='9' x='10.5' y='7' fill='%23979797' mask='url(%23b)' rx='1' transform='rotate(90 11.5 11.5)'/%3E %3Cpath fill='%23979797' fill-rule='nonzero' d='M11.5 21a9.5 9.5 0 1 0 0-19 9.5 9.5 0 0 0 0 19zm0 1C5.701 22 1 17.299 1 11.5S5.701 1 11.5 1 22 5.701 22 11.5 17.299 22 11.5 22z' mask='url(%23b)'/%3E %3C/g%3E %3C/svg%3E");
}
.reactist.range_input .range_btn:hover {
  cursor: pointer;
}
.reactist.range_input input[type=range] {
  -webkit-appearance: none;
  width: 200px;
  height: 30px;
}
.reactist.range_input input[type=range]:focus {
  outline: none;
}
.reactist.range_input input[type=range]::-webkit-slider-runnable-track {
  width: 200px;
  height: 2px;
  cursor: pointer;
  animate: 0.2s;
  background: #808080;
}
.reactist.range_input input[type=range]::-webkit-slider-thumb {
  box-sizing: border-box;
  border: 1px solid #3F82EF;
  height: 20px;
  width: 20px;
  border-radius: 50%;
  background: white;
  cursor: pointer;
  -webkit-appearance: none;
  margin-top: -9px;
}
.reactist.range_input input[type=range]:focus::-webkit-slider-runnable-track {
  background: #8d8d8d;
}
.reactist.range_input input[type=range]::-moz-range-track {
  width: 200px;
  height: 2px;
  cursor: pointer;
  animate: 0.2s;
  background: #808080;
}
.reactist.range_input input[type=range]::-moz-range-thumb {
  box-sizing: border-box;
  border: 1px solid #3F82EF;
  height: 20px;
  width: 20px;
  border-radius: 50%;
  background: white;
  cursor: pointer;
}
.reactist.range_input input[type=range]::-ms-track {
  background: transparent;
  border-color: transparent;
  border-width: 20px 0;
  color: transparent;
}
.reactist.range_input input[type=range]::-ms-fill-lower {
  width: 200px;
  height: 2px;
  cursor: pointer;
  animate: 0.2s;
  background: #737373;
}
.reactist.range_input input[type=range]::-ms-fill-upper {
  width: 200px;
  height: 2px;
  cursor: pointer;
  animate: 0.2s;
  background: #808080;
}
.reactist.range_input input[type=range]::-ms-thumb {
  box-sizing: border-box;
  border: 1px solid #3F82EF;
  height: 20px;
  width: 20px;
  border-radius: 50%;
  background: white;
  cursor: pointer;
}
.reactist.range_input input[type=range]:focus::-ms-fill-lower {
  background: #808080;
}
.reactist.range_input input[type=range]:focus::-ms-fill-upper {
  background: #8d8d8d;
}


/*# sourceMappingURL=RangeInput.map*/