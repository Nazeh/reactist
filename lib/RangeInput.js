!function(n,e){"object"==typeof exports&&"object"==typeof module?module.exports=e(require("prop-types"),require("react"),require("classnames")):"function"==typeof define&&define.amd?define("reactist",["prop-types","react","classnames"],e):"object"==typeof exports?exports.reactist=e(require("prop-types"),require("react"),require("classnames")):n.reactist=e(n["prop-types"],n.React,n.classnames)}(window,function(n,e,t){return function(n){var e={};function t(r){if(e[r])return e[r].exports;var i=e[r]={i:r,l:!1,exports:{}};return n[r].call(i.exports,i,i.exports,t),i.l=!0,i.exports}return t.m=n,t.c=e,t.d=function(n,e,r){t.o(n,e)||Object.defineProperty(n,e,{enumerable:!0,get:r})},t.r=function(n){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(n,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(n,"__esModule",{value:!0})},t.t=function(n,e){if(1&e&&(n=t(n)),8&e)return n;if(4&e&&"object"==typeof n&&n&&n.__esModule)return n;var r=Object.create(null);if(t.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:n}),2&e&&"string"!=typeof n)for(var i in n)t.d(r,i,function(e){return n[e]}.bind(null,i));return r},t.n=function(n){var e=n&&n.__esModule?function(){return n.default}:function(){return n};return t.d(e,"a",e),e},t.o=function(n,e){return Object.prototype.hasOwnProperty.call(n,e)},t.p="",t(t.s=50)}({0:function(e,t){e.exports=n},1:function(n,t){n.exports=e},14:function(n,e){n.exports=function(n){return"string"!=typeof n?n:(/^['"].*['"]$/.test(n)&&(n=n.slice(1,-1)),/["'() \t\n]/.test(n)?'"'+n.replace(/"/g,'\\"').replace(/\n/g,"\\n")+'"':n)}},2:function(n,e){n.exports=t},3:function(n,e){n.exports=function(n){var e=[];return e.toString=function(){return this.map(function(e){var t=function(n,e){var t=n[1]||"",r=n[3];if(!r)return t;if(e&&"function"==typeof btoa){var i=function(n){return"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(n))))+" */"}(r),o=r.sources.map(function(n){return"/*# sourceURL="+r.sourceRoot+n+" */"});return[t].concat(o).concat([i]).join("\n")}return[t].join("\n")}(e,n);return e[2]?"@media "+e[2]+"{"+t+"}":t}).join("")},e.i=function(n,t){"string"==typeof n&&(n=[[null,n,""]]);for(var r={},i=0;i<this.length;i++){var o=this[i][0];"number"==typeof o&&(r[o]=!0)}for(i=0;i<n.length;i++){var a=n[i];"number"==typeof a[0]&&r[a[0]]||(t&&!a[2]?a[2]=t:t&&(a[2]="("+a[2]+") and ("+t+")"),e.push(a))}},e}},4:function(n,e,t){var r={},i=function(n){var e;return function(){return void 0===e&&(e=n.apply(this,arguments)),e}}(function(){return window&&document&&document.all&&!window.atob}),o=function(n){var e={};return function(n,t){if("function"==typeof n)return n();if(void 0===e[n]){var r=function(n,e){return e?e.querySelector(n):document.querySelector(n)}.call(this,n,t);if(window.HTMLIFrameElement&&r instanceof window.HTMLIFrameElement)try{r=r.contentDocument.head}catch(n){r=null}e[n]=r}return e[n]}}(),a=null,s=0,u=[],c=t(5);function p(n,e){for(var t=0;t<n.length;t++){var i=n[t],o=r[i.id];if(o){o.refs++;for(var a=0;a<o.parts.length;a++)o.parts[a](i.parts[a]);for(;a<i.parts.length;a++)o.parts.push(b(i.parts[a],e))}else{var s=[];for(a=0;a<i.parts.length;a++)s.push(b(i.parts[a],e));r[i.id]={id:i.id,refs:1,parts:s}}}}function l(n,e){for(var t=[],r={},i=0;i<n.length;i++){var o=n[i],a=e.base?o[0]+e.base:o[0],s={css:o[1],media:o[2],sourceMap:o[3]};r[a]?r[a].parts.push(s):t.push(r[a]={id:a,parts:[s]})}return t}function f(n,e){var t=o(n.insertInto);if(!t)throw new Error("Couldn't find a style target. This probably means that the value for the 'insertInto' parameter is invalid.");var r=u[u.length-1];if("top"===n.insertAt)r?r.nextSibling?t.insertBefore(e,r.nextSibling):t.appendChild(e):t.insertBefore(e,t.firstChild),u.push(e);else if("bottom"===n.insertAt)t.appendChild(e);else{if("object"!=typeof n.insertAt||!n.insertAt.before)throw new Error("[Style Loader]\n\n Invalid value for parameter 'insertAt' ('options.insertAt') found.\n Must be 'top', 'bottom', or Object.\n (https://github.com/webpack-contrib/style-loader#insertat)\n");var i=o(n.insertAt.before,t);t.insertBefore(e,i)}}function d(n){if(null===n.parentNode)return!1;n.parentNode.removeChild(n);var e=u.indexOf(n);e>=0&&u.splice(e,1)}function g(n){var e=document.createElement("style");if(void 0===n.attrs.type&&(n.attrs.type="text/css"),void 0===n.attrs.nonce){var r=function(){0;return t.nc}();r&&(n.attrs.nonce=r)}return h(e,n.attrs),f(n,e),e}function h(n,e){Object.keys(e).forEach(function(t){n.setAttribute(t,e[t])})}function b(n,e){var t,r,i,o;if(e.transform&&n.css){if(!(o="function"==typeof e.transform?e.transform(n.css):e.transform.default(n.css)))return function(){};n.css=o}if(e.singleton){var u=s++;t=a||(a=g(e)),r=v.bind(null,t,u,!1),i=v.bind(null,t,u,!0)}else n.sourceMap&&"function"==typeof URL&&"function"==typeof URL.createObjectURL&&"function"==typeof URL.revokeObjectURL&&"function"==typeof Blob&&"function"==typeof btoa?(t=function(n){var e=document.createElement("link");return void 0===n.attrs.type&&(n.attrs.type="text/css"),n.attrs.rel="stylesheet",h(e,n.attrs),f(n,e),e}(e),r=function(n,e,t){var r=t.css,i=t.sourceMap,o=void 0===e.convertToAbsoluteUrls&&i;(e.convertToAbsoluteUrls||o)&&(r=c(r));i&&(r+="\n/*# sourceMappingURL=data:application/json;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(i))))+" */");var a=new Blob([r],{type:"text/css"}),s=n.href;n.href=URL.createObjectURL(a),s&&URL.revokeObjectURL(s)}.bind(null,t,e),i=function(){d(t),t.href&&URL.revokeObjectURL(t.href)}):(t=g(e),r=function(n,e){var t=e.css,r=e.media;r&&n.setAttribute("media",r);if(n.styleSheet)n.styleSheet.cssText=t;else{for(;n.firstChild;)n.removeChild(n.firstChild);n.appendChild(document.createTextNode(t))}}.bind(null,t),i=function(){d(t)});return r(n),function(e){if(e){if(e.css===n.css&&e.media===n.media&&e.sourceMap===n.sourceMap)return;r(n=e)}else i()}}n.exports=function(n,e){if("undefined"!=typeof DEBUG&&DEBUG&&"object"!=typeof document)throw new Error("The style-loader cannot be used in a non-browser environment");(e=e||{}).attrs="object"==typeof e.attrs?e.attrs:{},e.singleton||"boolean"==typeof e.singleton||(e.singleton=i()),e.insertInto||(e.insertInto="head"),e.insertAt||(e.insertAt="bottom");var t=l(n,e);return p(t,e),function(n){for(var i=[],o=0;o<t.length;o++){var a=t[o];(s=r[a.id]).refs--,i.push(s)}n&&p(l(n,e),e);for(o=0;o<i.length;o++){var s;if(0===(s=i[o]).refs){for(var u=0;u<s.parts.length;u++)s.parts[u]();delete r[s.id]}}}};var m=function(){var n=[];return function(e,t){return n[e]=t,n.filter(Boolean).join("\n")}}();function v(n,e,t,r){var i=t?"":r.css;if(n.styleSheet)n.styleSheet.cssText=m(e,i);else{var o=document.createTextNode(i),a=n.childNodes;a[e]&&n.removeChild(a[e]),a.length?n.insertBefore(o,a[e]):n.appendChild(o)}}},5:function(n,e){n.exports=function(n){var e="undefined"!=typeof window&&window.location;if(!e)throw new Error("fixUrls requires window.location");if(!n||"string"!=typeof n)return n;var t=e.protocol+"//"+e.host,r=t+e.pathname.replace(/\/[^\/]*$/,"/");return n.replace(/url\s*\(((?:[^)(]|\((?:[^)(]+|\([^)(]*\))*\))*)\)/gi,function(n,e){var i,o=e.trim().replace(/^"(.*)"$/,function(n,e){return e}).replace(/^'(.*)'$/,function(n,e){return e});return/^(#|data:|http:\/\/|https:\/\/|file:\/\/\/|\s*$)/i.test(o)?n:(i=0===o.indexOf("//")?o:0===o.indexOf("/")?t+o:r+o.replace(/^\.\//,""),"url("+JSON.stringify(i)+")")})}},50:function(n,e,t){"use strict";t.r(e);t(51);var r=t(1),i=t.n(r),o=t(0),a=t.n(o),s=t(2),u=t.n(s),c=function(n){var e=n.value,t=n.min,r=n.max,o=n.stepSize,a=n.onPlus,s=n.onMinus,c=n.onChange,p=n.className;"function"!=typeof a&&(a=c),"function"!=typeof s&&(s=c);var l=u()("reactist range_input",p);return i.a.createElement("div",{className:l},i.a.createElement("span",{className:"range_btn minus",onClick:function(){return e>t&&s(e-o)}}),i.a.createElement("input",{value:e,className:"range_slider",type:"range",min:t,max:r,step:o,onChange:function(n){return c(parseInt(n.target.value))}}),i.a.createElement("span",{className:"range_btn plus",onClick:function(){return e<r&&a(e+o)}}))};c.displayName="RangeInput",c.defaultProps={value:0,min:0,max:100,stepSize:1},c.propTypes={value:a.a.number.isRequired,min:a.a.number,max:a.a.number,stepSize:a.a.number,onPlus:a.a.func,onMinus:a.a.func,onChange:a.a.func.isRequired,className:a.a.string},e.default=c},51:function(n,e,t){var r=t(52);"string"==typeof r&&(r=[[n.i,r,""]]);var i={hmr:!0,transform:void 0,insertInto:void 0};t(4)(r,i);r.locals&&(n.exports=r.locals)},52:function(n,e,t){var r=t(14);(n.exports=t(3)(!1)).push([n.i,".reactist.range_input .range_btn {\n  height: 30px;\n  width: 30px;\n  background-size: 20px;\n  display: inline-block;\n  background-position: center;\n  background-repeat: no-repeat;\n}\n.reactist.range_input .range_btn.plus {\n  background-image: url("+r(t(53))+");\n}\n.reactist.range_input .range_btn.minus {\n  background-image: url("+r(t(54))+");\n}\n.reactist.range_input .range_btn:hover {\n  cursor: pointer;\n}\n.reactist.range_input input[type=range] {\n  -webkit-appearance: none;\n  width: 200px;\n  height: 30px;\n}\n.reactist.range_input input[type=range]:focus {\n  outline: none;\n}\n.reactist.range_input input[type=range]::-webkit-slider-runnable-track {\n  width: 200px;\n  height: 2px;\n  cursor: pointer;\n  animate: 0.2s;\n  background: #808080;\n}\n.reactist.range_input input[type=range]::-webkit-slider-thumb {\n  box-sizing: border-box;\n  border: 1px solid #3F82EF;\n  height: 20px;\n  width: 20px;\n  border-radius: 50%;\n  background: white;\n  cursor: pointer;\n  -webkit-appearance: none;\n  margin-top: -9px;\n}\n.reactist.range_input input[type=range]:focus::-webkit-slider-runnable-track {\n  background: #8d8d8d;\n}\n.reactist.range_input input[type=range]::-moz-range-track {\n  width: 200px;\n  height: 2px;\n  cursor: pointer;\n  animate: 0.2s;\n  background: #808080;\n}\n.reactist.range_input input[type=range]::-moz-range-thumb {\n  box-sizing: border-box;\n  border: 1px solid #3F82EF;\n  height: 20px;\n  width: 20px;\n  border-radius: 50%;\n  background: white;\n  cursor: pointer;\n}\n.reactist.range_input input[type=range]::-ms-track {\n  background: transparent;\n  border-color: transparent;\n  border-width: 20px 0;\n  color: transparent;\n}\n.reactist.range_input input[type=range]::-ms-fill-lower {\n  width: 200px;\n  height: 2px;\n  cursor: pointer;\n  animate: 0.2s;\n  background: #737373;\n}\n.reactist.range_input input[type=range]::-ms-fill-upper {\n  width: 200px;\n  height: 2px;\n  cursor: pointer;\n  animate: 0.2s;\n  background: #808080;\n}\n.reactist.range_input input[type=range]::-ms-thumb {\n  box-sizing: border-box;\n  border: 1px solid #3F82EF;\n  height: 20px;\n  width: 20px;\n  border-radius: 50%;\n  background: white;\n  cursor: pointer;\n}\n.reactist.range_input input[type=range]:focus::-ms-fill-lower {\n  background: #808080;\n}\n.reactist.range_input input[type=range]:focus::-ms-fill-upper {\n  background: #8d8d8d;\n}\n",""])},53:function(n,e){n.exports="\"data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' width='23' height='23' viewBox='0 0 23 23'%3E %3Cdefs%3E %3Cpath id='a' d='M0 0h23v23H0z'/%3E %3C/defs%3E %3Cg fill='none' fill-rule='evenodd'%3E %3Cmask id='b' fill='%23fff'%3E %3Cuse xlink:href='%23a'/%3E %3C/mask%3E %3Cpath fill='%23979797' fill-rule='nonzero' d='M11.5 21a9.5 9.5 0 1 0 0-19 9.5 9.5 0 0 0 0 19zm0 1C5.701 22 1 17.299 1 11.5S5.701 1 11.5 1 22 5.701 22 11.5 17.299 22 11.5 22z' mask='url(%23b)'/%3E %3Crect width='2' height='9' x='10.5' y='7' fill='%23979797' mask='url(%23b)' rx='1' transform='rotate(90 11.5 11.5)'/%3E %3Crect width='2' height='9' x='10.5' y='7' fill='%23979797' mask='url(%23b)' rx='1' transform='rotate(-180 11.5 11.5)'/%3E %3C/g%3E %3C/svg%3E\""},54:function(n,e){n.exports="\"data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' width='23' height='23' viewBox='0 0 23 23'%3E %3Cdefs%3E %3Cpath id='a' d='M0 0h23v23H0z'/%3E %3C/defs%3E %3Cg fill='none' fill-rule='evenodd'%3E %3Cmask id='b' fill='%23fff'%3E %3Cuse xlink:href='%23a'/%3E %3C/mask%3E %3Crect width='2' height='9' x='10.5' y='7' fill='%23979797' mask='url(%23b)' rx='1' transform='rotate(90 11.5 11.5)'/%3E %3Cpath fill='%23979797' fill-rule='nonzero' d='M11.5 21a9.5 9.5 0 1 0 0-19 9.5 9.5 0 0 0 0 19zm0 1C5.701 22 1 17.299 1 11.5S5.701 1 11.5 1 22 5.701 22 11.5 17.299 22 11.5 22z' mask='url(%23b)'/%3E %3C/g%3E %3C/svg%3E\""}})});
//# sourceMappingURL=RangeInput.map