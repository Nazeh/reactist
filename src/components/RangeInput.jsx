import './styles/range_input.less'

import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const RangeInput = ({
    value,
    min,
    max,
    stepSize,
    onPlus,
    onMinus,
    onChange,
    className
}) => {
    if (typeof onPlus !== 'function') {
        onPlus = onChange
    }
    if (typeof onMinus !== 'function') {
        onMinus = onChange
    }
    const rangeInputClassName = classNames('reactist_range_input', className)
    return (
        <div className={rangeInputClassName}>
            <span
                className="range_btn minus"
                onClick={() => value > min && onMinus(value - stepSize)}
            />
            <input
                value={value}
                className="range_slider"
                type="range"
                min={min}
                max={max}
                step={stepSize}
                onChange={event => onChange(parseInt(event.target.value))}
            />
            <span
                className="range_btn plus"
                onClick={() => value < max && onPlus(value + stepSize)}
            />
        </div>
    )
}
RangeInput.displayName = 'RangeInput'
RangeInput.defaultProps = {
    value: 0,
    min: 0,
    max: 100,
    stepSize: 1
}
RangeInput.propTypes = {
    /** Current value of the range input. */
    value: PropTypes.number.isRequired,
    /** Minimum value of the range input. */
    min: PropTypes.number,
    /** Maximum value of the range input. */
    max: PropTypes.number,
    /** Step size of the range input and the plus/minus buttons. */
    stepSize: PropTypes.number,
    /** Optional function that is called when plus button is clicked. If not supplied onChange will be called with the next value. */
    onPlus: PropTypes.func,
    /** Optional function that is called when minus button is clicked. If not supplied onChange will be called with the next value. */
    onMinus: PropTypes.func,
    /** Callback function that is called whenever the range input value changes. When onPlus or onMinus is supplied this will not be called for button clicks. */
    onChange: PropTypes.func.isRequired,
    /** Optional css class that is applied to the range input. */
    className: PropTypes.string
}

export default RangeInput
