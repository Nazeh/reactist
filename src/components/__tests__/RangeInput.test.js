import React from 'react'
import { shallow } from 'enzyme'

import RangeInput from '../RangeInput'

describe('RangeInput', () => {
    it('renders without crashing', () => {
        const rangeInput = shallow(getRangeInput())
        expect(rangeInput).toMatchSnapshot()
    })

    it('calls onChange when value of range input changes', () => {
        const onChangeSpy = jest.fn()
        const rangeInput = shallow(getRangeInput({ onChange: onChangeSpy }))

        rangeInput
            .find('.range_slider')
            .simulate('change', { target: { value: '60' } })
        expect(onChangeSpy).toHaveBeenLastCalledWith(60)
    })

    it('calls only onMinus when clicking on minus button', () => {
        const onMinusSpy = jest.fn()
        const onChangeSpy = jest.fn()
        const rangeInput = shallow(
            getRangeInput({ onMinus: onMinusSpy, onChange: onChangeSpy })
        )

        rangeInput.find('.range_btn.minus').simulate('click')
        expect(onMinusSpy).toHaveBeenLastCalledWith(49)
        expect(onChangeSpy).not.toHaveBeenCalled()
    })

    it('calls only onPlus when clicking on plus button', () => {
        const onPlusSpy = jest.fn()
        const onChangeSpy = jest.fn()
        const rangeInput = shallow(
            getRangeInput({ onPlus: onPlusSpy, onChange: onChangeSpy })
        )

        rangeInput.find('.range_btn.plus').simulate('click')
        expect(onPlusSpy).toHaveBeenLastCalledWith(51)
        expect(onChangeSpy).not.toHaveBeenCalled()
    })

    it('calls onChange if onMinus is not supplied when clicking on minus button', () => {
        const onChangeSpy = jest.fn()
        const rangeInput = shallow(getRangeInput({ onChange: onChangeSpy }))

        rangeInput.find('.range_btn.minus').simulate('click')
        expect(onChangeSpy).toHaveBeenLastCalledWith(49)
    })

    it('calls onChange if onPlus is not supplied when clicking on plus button', () => {
        const onChangeSpy = jest.fn()
        const rangeInput = shallow(getRangeInput({ onChange: onChangeSpy }))

        rangeInput.find('.range_btn.plus').simulate('click')
        expect(onChangeSpy).toHaveBeenLastCalledWith(51)
    })

    it('does not call onPlus if the value is at the maximum', () => {
        const onPlusSpy = jest.fn()
        const rangeInput = shallow(
            getRangeInput({ onPlus: onPlusSpy, value: 100 })
        )

        rangeInput.find('.range_btn.plus').simulate('click')
        expect(onPlusSpy).not.toHaveBeenCalled()
    })

    it('does not call onMinus if the value is at the minimum', () => {
        const onMinusSpy = jest.fn()
        const rangeInput = shallow(
            getRangeInput({ onMinus: onMinusSpy, value: 0 })
        )

        rangeInput.find('.range_btn.minus').simulate('click')
        expect(onMinusSpy).not.toHaveBeenCalled()
    })

    // helpers ================================================================
    const getRangeInput = props => (
        <RangeInput value={50} onChange={jest.fn()} {...props} />
    )
})
