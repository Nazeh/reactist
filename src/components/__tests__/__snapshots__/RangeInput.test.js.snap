// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeInput renders without crashing 1`] = `
<div
  className="reactist_range_input"
>
  <span
    className="range_btn minus"
    onClick={[Function]}
  />
  <input
    className="range_slider"
    max={100}
    min={0}
    onChange={[Function]}
    step={1}
    type="range"
    value={50}
  />
  <span
    className="range_btn plus"
    onClick={[Function]}
  />
</div>
`;
